lib/pg_ha_migrations/safe_statements.rb

```diff
@@ -286,6 +294,141 @@ def safe_create_partitioned_table(table, key:, type:, infer_primary_key: nil, **
     end
   end
 
+  def safe_partman_create_parent(table, **options)
+    if options[:retention].present? || options[:retention_keep_table] == false
+      raise PgHaMigrations::UnsafeMigrationError.new(":retention and/or :retention_keep_table => false can potentially result in data loss if misconfigured. Please use unsafe_partman_create_parent if you want to set these options")
+    end
+
+    unsafe_partman_create_parent(table, **options)
+  end
+
+  def unsafe_partman_create_parent(
+    table,
+    key:,
+    interval:,
+    infinite_time_partitions: true,
+    inherit_privileges: true,
+    premake: nil,
+    start_partition: nil,
+    template_table: nil,
+    retention: nil,
+    retention_keep_table: nil
+  )
+    raise ArgumentError, "Expected <key> to be present" unless key.present?
+    raise ArgumentError, "Expected <interval> to be present" unless interval.present?
+
+    if ActiveRecord::Base.connection.postgresql_version < 11_00_00
+      raise PgHaMigrations::InvalidMigrationError, "Native partitioning with partman not supported on Postgres databases before version 11"
+    end
+
+    create_parent_options = {
+      parent_table: _fully_qualified_table_name_for_partman(table),
+      template_table: template_table ? _fully_qualified_table_name_for_partman(template_table) : nil,
+      control: key,
+      type: "native",
+      interval: interval,
+      premake: premake,
+      start_partition: start_partition,
+    }.compact
+
+    create_parent_sql = create_parent_options.map { |k, v| "p_#{k} := #{connection.quote(v)}" }.join(", ")
+
+    log_message = "partman_create_parent(#{table}, " \
```

Comment:
Should we use `<<~EOS` style for ease of reading?

Reply:
I mean maybe, but that introduces newline characters so we'd have to split on `\n` and then join on `, `

Reply:
So the heredoc form can also have the `\` right before the line breaks, FWIW.